pkg/openstack/placement.go

```diff
@@ -55,6 +55,12 @@ func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackC
 		}
 	}
 
+	// set CA cert and preserve any previously set TLS certs
+	if instance.Spec.TLS.Enabled(service.EndpointInternal) {
+		instance.Spec.Placement.Template.TLS = placementAPI.Spec.TLS
+	}
+	instance.Spec.Placement.Template.TLS.CaBundleSecretName = instance.Status.TLS.CaBundleSecretName
```

Comment:
It is a bit strange that we are updating our own input fields (i.e. Spec) but if it is not causing any trouble with the controller-runtime / k8s then fine.

Reply:
we are not patching the CR and writing the spec in our own. the defer function is only updating metadata and status[1] . Setting it here on the instance because we copy the template to the placement spec `instance.Spec.Placement.Template.DeepCopyInto(&placementAPI.Spec)` 

[1] https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/helper/helper.go#L193

Reply:
Oh cool then. Thanks for the explanation.